tutorials/xr/openxr_spatial_entities.rst

```diff
@@ -0,0 +1,1738 @@
+                    # Use marker_tracker.marker_id to identify the marker.
+                    pass
+
+As we can see, QR Codes provide a data block that is either a string or a byte array.
+Aruco and April tags provide an ID that is read from the code.
+
+It's up to your use case how best to link the marker data to the scene that needs to be loaded.
+An example would be to encode the name of the asset you wish to display in a QR code.
+
+**Maybe see if we can work out at least one example here, need access to hardware that supports this, should have that soon**
+
+Backend access
+--------------
+
+For most purposes the core system, along with any vendor extensions, should be what most
+users would use as provided.
+
+For those who are implementing vendor extensions, or those for who the built-in logic doesn't
+suffice, backend access is provided through a set of singleton objects.
+
+These objects can also be used to query what capabilities are supported by the headset in use.
+We've already added code that checks for these in our spatial manager and spatial anchor code
+in the sections above.
+
+.. note::
+
+    The spatial entities system will encapsulate many OpenXR entities in resources that are
+    returned as RIDs.
+
+Spatial entity core
+~~~~~~~~~~~~~~~~~~~
+
+The core spatial entity functionality is exposed through the 
+:ref:`OpenXRSpatialEntityExtension<class_OpenXRSpatialEntityExtension>` singleton.
+
+Specific logic is exposed through capabilities that introduce specialised component types,
+and give access to specific types of entities, however they all use the same mechanisms
+for accessing the entity data managed by the spatial entity system.
+
+We'll start by having a look at the individual components that make up the core system.
+
+Spatial contexts
+""""""""""""""""
+
+A spatial context is the main object through which we query the spatial entities system.
+Spatial contexts allow us to configure how we interact with one or more capabilities.
+
+It's recommended to create a spatial context for each capability that you wish to interact
+with, in fact, this is what Godot does for its built-in logic.
+
+We start by setting the capability configuration objects for the capabilities we wish to
+access.
+Each capability will enable the components we support for that capability.
+Settings can determine which components will be enabled.
+We'll look at these configuration objects in more detail as we look at each supported capability.
+
+Creating a spatial context is an asynchronous action. This means we ask the XR runtime to
+create a spatial context, and at a point in the future the XR runtime will provide us
+with the result.
+
+The following script is the start of our example and can be added as a node to your scene.
+It shows the creation of a spatial context for plane tracking,
+and sets up our entity discovery.
+
+.. code-block:: gdscript
+
+    extends Node
+
+    var spatial_context : RID
```

Comment:
I think I'm going to stop pointing these out. They can all be grep'd for and there's also the possibility that the docs style doesn't mandate this anyway :-)

```suggestion
    var spatial_context: RID
```